### DISCUSSION TITLE
QRemeshify 1.0.0

### DISCUSSION THREAD

**Discussion by ksami** [ksami/QRemeshify#1] [Announcements]
Refer to [README.md](https://github.com/ksami/QRemeshify/blob/main/README.md) for docs

<hr /><em>This discussion was created from the release <a href='https://github.com/ksami/QRemeshify/releases/tag/1.0.0'>QRemeshify 1.0.0</a>.</em>

**Comment by MetinSeven** [comment]
Thanks for developing and sharing this add-on, much appreciated. You can find some first tests here:

- https://blenderartists.org/t/quad-remesher-auto-retopologizer/1170913/634
- https://blenderartists.org/t/the-big-blender-sculpt-mode-thread-part-2/1395490/3765

It seems that QRemeshify is very good in handling organic meshes, and less effective on hard-surface meshes, but these are just some first tests.

**Comment by Denyalife** [comment]
Thanks for the addon. The properties description says, guide edge flow with edges marked seams/sharp, but it doesn't work for me no matter what I try, Sharp Detect option does not respond sharp, seams edges. It would be nice to add an option to save the edge by material and Face Sets. Scale factor is a very important option and I think it should be moved to the main settings.
You are cool, your work will be useful to many

**Reply by ksami** [reply]
Took a quick look and I think there might be a bug with the edges marked sharp, opened #4 to investigate further

**Reply by ksami** [reply]
Regarding saving the edge by material and face sets, if the same method is used as with edges marked seam/sharp, though it helps with guiding the edge flow it does also affect the overall shape of the mesh, and I'm not sure if that's a desirable result.

Can I know why you think scale factor is important? To me it looks like it just applies a simple subdivision after remeshing.

**Reply by ksami** [reply]
Fixed #4 I must have been dreaming that it ever worked, marking as sharp should have a strong influence on the edge flow now

**Reply by MetinSeven** [reply]
Thanks! 👍

**Reply by MetinSeven** [reply]
Works better now! 👍

![Blender remesher add-ons test C](https://github.com/user-attachments/assets/dba5cf0a-9e6f-4188-8e9d-742a62edc08c)

**Reply by Denyalife** [reply]
Preserving the material boundary, Face Sates 
This parameter should be optional. In my experience Quad Remesher sometimes gives better results with material boundary tracking than just marking the same edges as sharp. Also, in some cases it can be faster and more convenient, especially if the model already contains materials, for example if you imported a model from Marvelous Designer. There can be several materials and now we have many sets of edges that we can select, delete, combine with the same sharp edges and choose the best one. And of course we can reassign the newly created areas to the materials, it is useful and can help to analyze the shape and be face sets for sculpting.
Here is another example. Sculpt the shape, create Face Sets, smooth the edges of Face Sets, and generate a new mesh with saved edges of Face Sets. Then we can set thickness, chamfer... This is a simple example of many and a basis for solid modeling. 
In fact, in high-poly heavy models this will be the only possible way to set directions.

https://www.youtube.com/watch?v=6K0lgTd6KiM&t=2s

![1](https://github.com/user-attachments/assets/2e7a3dba-277f-4237-a88c-1b4699a8d13f)
![2](https://github.com/user-attachments/assets/eea808d2-d931-4084-bf74-10c7aa37354e)
![3](https://github.com/user-attachments/assets/e28fee14-b32e-4fca-9779-51a4b98f5736)
![4](https://github.com/user-attachments/assets/b67b06f1-8665-4b90-82a2-d7abe56b9a9a)
![5](https://github.com/user-attachments/assets/dc5e0d08-db7e-4f3b-8d8b-aab6b36ae4df)
![6](https://github.com/user-attachments/assets/c4644527-bedd-4afc-bd71-904292154998)

**Reply by Denyalife** [reply]
Scale Factor
Scale Factor is not a simple subdivision and is very important for the final number of polygons. The larger the number, the larger the polygon size. It would be much more convenient to control not the polygon size, but the approximate total number of polygons.
In a medium density mesh, the Preprocess option works well. But with complex folds penetrating inside itself, the detail is lost. With Preprocess turned off, the result is good, but with a very dense mesh. This can be corrected with Scale Factor 3. But with a too high Scale Factor value and Preprocess disabled, the result is worse than with Preprocess enabled and Scale Factor 1. I have attached  .blend

https://drive.google.com/file/d/1uLLkSgVmAVizVkbVcp0d92ubenuAMg0l/view?usp=sharing
![111](https://github.com/user-attachments/assets/ee04fc2e-4dca-4e7d-8c80-a10f9be7b35a)

**Reply by Denyalife** [reply]
Also Alpha is defaulted to 0.005 and it works fine. But, as in other retopology applications, tightening appear in places like holes. This can be fixed by changing the Alpha parameter to a higher value, but then loops may appear.
![11c37d885b5de9314497393b5c31293612888d95 copy](https://github.com/user-attachments/assets/c4886f6f-b010-4969-b235-3830b82690a3)

I would like to be able to set different Alpha values for different vertices. The same applies to the Scale Factor parameter, there are more details, the denser the mesh. 
If the original object is not in the center of the coordinates, the new generated one jumps to the center of the scene. This is not convenient.
I appreciate what you have already done, thank you.

**Reply by ksami** [reply]
I'm unfamiliar with such workflows so this is helpful thanks, I'll try looking into material boundary and face sets.

Yeah I agree scale factor looks important in that case, I only tried scale factor <1 that's why I thought it was just subdivision 🤦 

Setting alpha and scale factor per vertex is not possible though as that functionality is not exposed from QuadWild, what you might be looking for could be the parameters for the field computation, the [.rosy file described here](https://github.com/nicopietroni/quadwild?tab=readme-ov-file#quadwild), but that's way out of my depth.

Setting the object location back is being implemented thanks to #6

**Reply by Denyalife** [reply]
Thanks for your work. Have a nice weekend

**Comment by Denyalife** [comment]
In case you haven't heard
https://extensions.blender.org/

**Reply by MetinSeven** [reply]
It'd be great (and easier to install and update) if QRemeshify would become an official extension. 👍

**Reply by ksami** [reply]
I'm currently exploring doing this for a living and unfortunately the official extensions platform do not support that, perhaps if another extensions platform comes up.
